### PR TITLE
test: make try_merge reliable

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: gengliqi <gengliqiii@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/8506

Problem Summary:
Fix unstable test `test_node_mutiple_rollback_merge`.
Because of https://github.com/tikv/tikv/pull/8441, the admin request will be rejected if this peer hasn't applied to current term.

### What is changed and how it works?

What's Changed:
Retry request in `call_command_on_leader` if the response gets this error.
Add `must_try_merge` used for verfying `try_merge` must succeed.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

No

### Release note 
* No release note

### CODE REVIEW

File: components/test_raftstore/src/cluster.rs [hunk 1]
```diff
@@ -1255,12 +1255,46 @@ impl<T: Simulator> Cluster<T> {
...
+            source_region.get_id(),
+            source_region.get_region_epoch(),
+            prepare_merge,
+        );
+        let mut try_cnt = 0;
+        loop {
+            let resp = self
+                .call_command_on_leader(req.clone(), Duration::from_secs(3))
+                .unwrap();
+            if !resp
```
Comment (BusyJay) [inline components/test_raftstore/src/cluster.rs:1271]: Can it be moved into `call_command_on_leader`? I think other commands can be affected too.

**Review by BusyJay — APPROVED** [review]

**Comment by NingLin-P** [comment]
Instead of `must_try_merge` how about just `must_merge`?

**Comment by gengliqi** [comment]
> Instead of `must_try_merge` how about just `must_merge`?

The `must_try_merge` is async while `must_merge` is sync which means it returns until the merge process succeeds.

**Review by NingLin-P — APPROVED** [review]

**Pull request merged**